src/main/java/teammates/ui/webapi/action/UpdateInstructorPrivilegeAction.java

```diff
@@ -0,0 +1,174 @@
+        response.setCanModifyCourse(instructorToUpdate.privileges.isAllowedForPrivilege(
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_COURSE));
+
+        if (sessionName == null && sectionName == null) {
+            response.setCanViewStudentInSections(instructorToUpdate.privileges.isAllowedForPrivilege(
+                    Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS));
+            response.setCanViewSessionInSections(instructorToUpdate.privileges.isAllowedForPrivilege(
+                    Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS));
+            response.setCanSubmitSessionInSections(instructorToUpdate.privileges.isAllowedForPrivilege(
+                    Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS));
+            response.setCanModifySessionCommentsInSections(instructorToUpdate.privileges.isAllowedForPrivilege(
+                    Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION_COMMENT_IN_SECTIONS));
+        } else {
+            response.setCanViewStudentInSections(instructorToUpdate.privileges.isAllowedForPrivilege(
+                    sectionName, Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS));
+
+            if (sessionName == null) {
+                response.setCanViewSessionInSections(instructorToUpdate.privileges.isAllowedForPrivilege(
+                        sectionName, Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS));
+                response.setCanSubmitSessionInSections(instructorToUpdate.privileges.isAllowedForPrivilege(
+                        sectionName, Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS));
+                response.setCanModifySessionCommentsInSections(instructorToUpdate.privileges.isAllowedForPrivilege(
+                        sectionName, Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION_COMMENT_IN_SECTIONS));
+            } else {
+                response.setCanViewSessionInSections(instructorToUpdate.privileges.isAllowedForPrivilege(
+                        sectionName, sessionName, Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS));
+                response.setCanSubmitSessionInSections(instructorToUpdate.privileges.isAllowedForPrivilege(
+                        sectionName, sessionName, Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS));
+                response.setCanModifySessionCommentsInSections(
+                        instructorToUpdate.privileges.isAllowedForPrivilege(sectionName, sessionName,
+                                Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION_COMMENT_IN_SECTIONS));
+            }
+        }
+
+        return new JsonResult(response);
+    }
+
+    private void updateCourseLevelPrivileges(Map<String, Boolean> privilegesMap, InstructorAttributes toUpdate) {
+        for (HashMap.Entry<String, Boolean> entry : privilegesMap.entrySet()) {
```

Comment:
Prefer interface instead of concrete type.

```suggestion
        for (Map.Entry<String, Boolean> entry : privilegesMap.entrySet()) {
```

Same in other places